src/NzbDrone.Core/MediaFiles/ScriptImportDecider.cs

```diff
@@ -22,18 +24,21 @@ public class ImportScriptService : IImportScript
         private readonly IVideoFileInfoReader _videoFileInfoReader;
         private readonly IProcessProvider _processProvider;
         private readonly IConfigService _configService;
+        private readonly ITagRepository _tagRepository;
         private readonly Logger _logger;
 
         public ImportScriptService(IProcessProvider processProvider,
                                    IVideoFileInfoReader videoFileInfoReader,
                                    IConfigService configService,
                                    IConfigFileProvider configFileProvider,
+                                   ITagRepository tagRepository,
                                    Logger logger)
         {
             _processProvider = processProvider;
             _videoFileInfoReader = videoFileInfoReader;
             _configService = configService;
             _configFileProvider = configFileProvider;
+            _tagRepository = tagRepository;
```

Comment:
This should use the `TagService` instead of `TagRepository` (same for other uses).

Reply:
That's what I tried first, but that actually gives a dependency loops because the TagService was involved in sending notifications, but obviously notifications depend on the various custom scripts, so those can't depend on the TagService. Using the repository directly breaks that dependency cycle.